src/Linq/Extensions.cs

```diff
@@ -248,6 +242,29 @@ public static IList<T> Shuffled<T>(this IEnumerable<T> source, Random random = n
             return list;
         }
 
+        /// <summary>
+        /// Returns a new shuffled list with the elements from the given sequence.
+        /// </summary>
+        /// <param name="source">The sequence to shuffle.</param>
+        /// <param name="random">An optional random object to be used. If none is given, StaticRandom is used instead.</param>
+        public static IEnumerable<T> ShuffledDeferred<T>(this IEnumerable<T> source, Random random = null)
```

Comment:
To me it is not entirely clear what this method does. I.e. what is the difference between this and just shuffling. For that reason, I think the comments have to be slightly more extensive.